lab-jaren/lib/storage.js

```diff
@@ -0,0 +1,55 @@
+'use strict';
+
+const Promise = require('bluebird');
+const fs = Promise.promisifyAll(require('fs'));
+
+const storage = module.exports = {};
+
+const dataDir = `${__dirname}/../data`;
+
+storage.createItem = function(name, item) {
+  if (!(`${dataDir}/${name}`)) {
```

Comment:
I just noticed this if block here and in your ``storage.readItem`` method too. This actually isn't working because ``mkdir`` is an npm package you need to install as a dependncy and require. 

There are several different versions. Even though it's relatively new, I'd recommend [mkdirp-bluebird](https://www.npmjs.com/package/mkdirp-bluebird) because it's already promisified for you. This means you **do not** need to append the word ``Async`` to it as it's already a promise.

Then again, this npm package is also fairly new and doesn't have a lot of downloads yet (which goes against Duncan's advice that we should look at npm packages with lots of downloads as more fail-safe), but I suspect it'll grow in popularity very soon since it's new. 

If you choose to go with other ``mkdirp`` npms, read their docs about whether they're already promisified to see if you need to append the word ``Async`` to it manually (or just make your API requests and see if it breaks, I'm sure you'll figure out which is right).

Also, since we already have a ``.then/.catch`` promise chain happening, there's no need for if statements. 

You would start your chain with the ``return fs.StatAsync(...)`` method, and then the first event chained would be a ``catch`` block that invokes your ``mkdirp`` function to create the folder if the folder doesn't already exist. 

You could nest a ``try/catch`` block inside here to handle any errors with ``mkdir`` itself.  

Upon successful creation of your folders, you'd move on to a ``then`` block that writes the actually file with the stringified JSON, then you'll move on to the next item to return the item so your server route can get it. 

This is a lot, but I hope it makes sense. Let me know if you have questions.